encoding/bufpool/bufpool.go

```diff
@@ -0,0 +1,22 @@
+// Package bufpool is a freelist for bytes.Buffer objects.
+package bufpool
+
+import (
+	"bytes"
+	"sync"
+)
+
+var pool = &sync.Pool{New: func() interface{} { return bytes.NewBuffer(nil) }}
```

Comment:
Maybe consider `new(bytes.Buffer)` which produces the same result with less machinery.